replica/distributed_loader.cc

```diff
@@ -451,11 +451,18 @@ distributed_loader::process_upload_dir(distributed<replica::database>& db, distr
     return seastar::async(std::move(attr), [&db, &view_update_generator, &sys_dist_ks, ks = std::move(ks), cf = std::move(cf)] {
         global_column_family_ptr global_table(db, ks, cf);
 
+        sharded<locator::effective_replication_map_ptr> erms;
+        erms.start(sharded_parameter([&global_table] {
+            return global_table->get_effective_replication_map();
+        })).get();
+        auto stop_erms = deferred_stop(erms);
```

Comment:
This could be done in another way: instead of delegating to the sstable and asking into to look at the owner shards, we could intersect the [first, last] token range with the tablet map. This could tell us not only whether the sstable is owned by multiple shards (and therefore needs immediate splitting) but also whether it's owned by multiple tablets (and needs non-immediate splitting).

Options:
1. keep as is
2. abort() for now
3. move the checking code here

Reply:
It won't work now when sstables are shared by many tables, since [first, last] range can overlap with tablets from other shards, even though the sstable itself contains only ranges of tablets from a single shard.